src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Common/SyntaxNodeExtensions.cs

```diff
@@ -303,5 +303,120 @@ public static StringComparer GetStringComparer(this SyntaxNode? node)
+                || (parentNode is VBSyntax.MemberAccessExpressionSyntax vbMAE && vbMAE.Expression.IsEquivalentTo(node)))
+            {
+                return true;
+            }
+
+            // Base type
+            if (parentNode.IsBaseTypeSyntax())
+            {
+                return true;
+            }
+
+            // Generic parameter
+            if (parentNode.IsKind(CS.SyntaxKind.TypeArgumentList) || parentNode.IsKind(VB.SyntaxKind.TypeArgumentList))
+            {
+                return true;
+            }
+
+            // Attribute
+            if (parentNode.IsKind(CS.SyntaxKind.Attribute) || parentNode.IsKind(VB.SyntaxKind.Attribute))
+            {
+                return true;
+            }
+
+            // Array
+            if (parentNode.IsKind(CS.SyntaxKind.ArrayType) || parentNode.IsKind(VB.SyntaxKind.ArrayType))
+            {
+                return true;
+            }
+
+            // Variable type
+            if ((parentNode is CSSyntax.VariableDeclarationSyntax csDecl && csDecl.Type.IsEquivalentTo(node))
+                || (parentNode is VBSyntax.SimpleAsClauseSyntax vbDecl && vbDecl.Type.IsEquivalentTo(node)))
+            {
+                return true;
+            }
+
+            // Parameter type
+            if (parentNode.IsKind(CS.SyntaxKind.Parameter) || parentNode.IsKind(VB.SyntaxKind.Parameter))
+            {
+                return true;
+            }
+
+            // Property type
+            if (parentNode.IsKind(CS.SyntaxKind.PropertyDeclaration) || parentNode.IsKind(VB.SyntaxKind.PropertyStatement))
+            {
+                return true;
+            }
+
+            // Method return type (VB method return types are already covered by SimpleAsClause
+            if (parentNode.IsKind(CS.SyntaxKind.MethodDeclaration))
+            {
+                return true;
+            }
+
+            // C# As expression
+            if (parentNode is CSSyntax.BinaryExpressionSyntax binaryExpression
+                && binaryExpression.IsKind(CS.SyntaxKind.AsExpression)
+                && binaryExpression.Right.IsEquivalentTo(node))
+            {
+                return true;
+            }
+
+            // VB CType expression
+            if (parentNode is VBSyntax.CTypeExpressionSyntax cType
+                && cType.Type.IsEquivalentTo(node))
+            {
+                return true;
+            }
+
+            // C# cast expression
+            if (parentNode is CSSyntax.CastExpressionSyntax csCastExpression
+                && csCastExpression.Type.IsEquivalentTo(node))
+            {
+                return true;
+            }
+
+            // VB cast expression
+            if (parentNode is VBSyntax.CastExpressionSyntax vbCastExpression
+                && vbCastExpression.Type.IsEquivalentTo(node))
+            {
+                return true;
+            }
+
+            return false;
```

Comment:
I can't think of any others, but we'll just need to keep an eye out for additional type-like syntaxes :)